sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py

```diff
@@ -98,20 +107,24 @@ def send(self, events, **kwargs):
         :raises: :class:`ValueError`, when events do not follow specified SendType.
          """
         if not isinstance(events, list):
-            events = [events]
+            events = cast(ListEventType, [events])
 
-        if all(isinstance(e, CloudEvent) for e in events) or all(_is_cloud_event(e) for e in events):
+        if all(isinstance(e, CloudEvent) for e in events) or all(_is_cloud_event(cast(Dict, e)) for e in events):
             try:
-                events = [e._to_generated(**kwargs) for e in events] # pylint: disable=protected-access
+                events = [cast(CloudEvent, e)._to_generated(**kwargs) for e in events] # pylint: disable=protected-access
             except AttributeError:
                 pass # means it's a dictionary
             kwargs.setdefault("content_type", "application/cloudevents-batch+json; charset=utf-8")
-            self._client.publish_cloud_event_events(self._topic_hostname, events, **kwargs)
+            self._client.publish_cloud_event_events(
+                self._topic_hostname,
+                cast(List[InternalCloudEvent], events),
+                **kwargs
+                )
         elif all(isinstance(e, EventGridEvent) for e in events) or all(isinstance(e, dict) for e in events):
             kwargs.setdefault("content_type", "application/json; charset=utf-8")
-            self._client.publish_events(self._topic_hostname, events, **kwargs)
+            self._client.publish_events(self._topic_hostname, cast(List[InternalEventGridEvent], events), **kwargs)
         elif all(isinstance(e, CustomEvent) for e in events):
-            serialized_events = [dict(e) for e in events]
-            self._client.publish_custom_event_events(self._topic_hostname, serialized_events, **kwargs)
+            serialized_events = [dict(e) for e in events] # type: ignore
```

Comment:
Why? What's the error?

Reply:
`Argument 1 to "dict" has incompatible type "Union[CloudEvent, EventGridEvent, CustomEvent, Dict[Any, Any]]"; expected "Mapping[Any, Any]"`
I can use a cast, but it won't entirely be true - ideally we validate that it's not a cloudevent, eventgrid event by the time we hit this line and they should not be included in the union - afaik, it's a problem with mypy.